src/proof_system/proof.rs

```diff
@@ -449,61 +467,75 @@ impl<E: PairingEngine, P: TEModelParameters<BaseField = E::Fr>> Proof<E, P> {
                 self.z_comm.0,
             );
 
-        let scalars_repr: Vec<<E::Fr as PrimeField>::BigInt> =
-            scalars.iter().map(|s| s.into_repr()).collect();
+        let scalars_repr =
+            scalars.iter().map(E::Fr::into_repr).collect::<Vec<_>>();
 
         Commitment(
             VariableBaseMSM::multi_scalar_mul(&points, &scalars_repr).into(),
         )
     }
 }
 
-// The first lagrange polynomial has the expression
-// L_0(X) = mul_from_1_to_(n-1) [(X - g^i) / (1 - g^i)]
-// with g being the generator of the domain (the nth root of unity)
-// We use two equalities:
-//   1. mul_from_2_to_(n-1)[ 1 / (1 - g^i)]  =  1 / n
-//   2. mul_from_2_to_(n-1)[ (X - g^i)] = (X^n -1) / (X - 1)
-// to obtain the expression:
-// L_0(X) = (X^n -1) / n * (X -1)
-fn compute_first_lagrange_evaluation<F: PrimeField>(
+/// The first lagrange polynomial has the expression:
+///
+/// ```text
+/// L_0(X) = mul_from_1_to_(n-1) [(X - g^i) / (1 - g^i)]
+/// ```
+///
+/// with `g` being the generator of the domain (the `n`th root of unity).
```

Comment:
We should have consistency naming the roots of unity, I think `omega` is the clearest option